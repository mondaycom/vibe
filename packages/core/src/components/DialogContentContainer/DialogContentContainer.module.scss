.dialogContentContainer:focus {
  outline: none;
}

.sizeSmall {
  padding: var(--spacing-small);
}

.sizeMedium {
  padding: var(--spacing-medium);
}

.sizeLarge {
  padding: var(--spacing-large);
}

.typePopover {
  box-shadow: var(--box-shadow-medium);
  border-radius: var(--border-radius-medium);
  background-color: var(--secondary-background-color);

  :global(.dark-app-theme) &,
  :global(.black-app-theme) &,
  :global(.hacker-app-theme) & {
    box-shadow: 0 0 0 1px var(--layout-border-color), var(--box-shadow-medium);
  }
}

.typeModal {
  box-shadow: var(--box-shadow-large);
  border-radius: var(--border-radius-big);
  background-color: var(--primary-background-color);
}
