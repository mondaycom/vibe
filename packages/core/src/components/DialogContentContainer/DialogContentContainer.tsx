import { camelCase } from "lodash-es";
import { getStyle } from "../../helpers/typesciptCssModulesHelper";
import cx from "classnames";
import React, { useRef, forwardRef } from "react";
import useMergeRef from "../../hooks/useMergeRef";
import { DialogSize as DialogSizeEnum, DialogType as DialogTypeEnum } from "../Dialog/DialogConstants";
import { DialogSize, DialogType } from "../Dialog";
import { withStaticProps, VibeComponentProps } from "../../types";
import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
import styles from "./DialogContentContainer.module.scss";

export interface DialogContentContainerProps extends VibeComponentProps {
  /**
   * The content inside the dialog container.
   */
  children?: React.ReactNode;
  /**
   * The ID of the element that labels this dialog.
   */
  ariaLabelledby?: string;
  /**
   * The ID of the element that describes this dialog.
   */
  ariaDescribedby?: string;
  /**
   * The type of dialog.
   */
  type?: DialogType;
  /**
   * The size of the dialog.
   */
  size?: DialogSize;
  /**
   * Custom styles applied to the dialog container.
   */
  style?: React.CSSProperties;
}

const DialogContentContainer = forwardRef(
  (
    {
      id,
      className = "",
      ariaLabelledby = "",
      ariaDescribedby = "",
      type = "popover",
      size = "small",
      children,
      style,
      "data-testid": dataTestId = getTestId(ComponentDefaultTestId.DIALOG_CONTENT_CONTAINER, id),
      ...props
    }: DialogContentContainerProps,
    ref: React.Ref<HTMLElement>
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    return (
      <div
        role="dialog"
        data-testid={dataTestId}
        aria-labelledby={ariaLabelledby}
        aria-describedby={ariaDescribedby}
        ref={mergedRef}
        style={style}
        className={cx(
          styles.dialogContentContainer,
          className,
          getStyle(styles, camelCase("type--" + type)),
          getStyle(styles, camelCase("size--" + size))
        )}
        {...props}
      >
        {children}
      </div>
    );
  }
);

interface DialogContentContainerStaticProps {
  types: typeof DialogTypeEnum;
  sizes: typeof DialogSizeEnum;
}

export default withStaticProps<DialogContentContainerProps, DialogContentContainerStaticProps>(DialogContentContainer, {
  types: DialogTypeEnum,
  sizes: DialogSizeEnum
});
