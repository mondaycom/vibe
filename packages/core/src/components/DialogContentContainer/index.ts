export { default as DialogContentContainer, type DialogContentContainerProps } from "./DialogContentContainer";
