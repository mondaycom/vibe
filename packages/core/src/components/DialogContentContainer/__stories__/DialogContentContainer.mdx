import { Meta } from "@storybook/blocks";
import * as DialogContentContainerStories from "./DialogContentContainer.stories";

<Meta of={DialogContentContainerStories} />

# DialogContentContainer

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Feedback](#feedback)

## Overview

This component is a style component, it provide the look and feel of elevation.

<Canvas of={DialogContentContainerStories.Overview} />

### Import

```js
import { DialogContentContainer } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines guidelines={["Use this component in order to wrap components within Dialog or Modal"]} />

## Variants

### Popover

<Canvas of={DialogContentContainerStories.Popover} />

### Modal

<Canvas of={DialogContentContainerStories.Modal} />
