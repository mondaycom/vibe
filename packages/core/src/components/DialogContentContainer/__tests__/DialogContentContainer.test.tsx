import { it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import DialogContentContainer from "../DialogContentContainer";

it("renders correctly with empty props", () => {
  const tree = renderer.create(<DialogContentContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with data-testid", () => {
  const tree = renderer.create(<DialogContentContainer data-testid="test" />).toJSON();
  expect(tree).toMatchSnapshot();
});
