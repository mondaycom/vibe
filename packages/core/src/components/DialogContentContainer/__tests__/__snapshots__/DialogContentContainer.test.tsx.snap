// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly with data-testid 1`] = `
<div
  aria-describedby=""
  aria-labelledby=""
  className="dialogContentContainer typePopover sizeSmall"
  data-testid="test"
  role="dialog"
/>
`;

exports[`renders correctly with empty props 1`] = `
<div
  aria-describedby=""
  aria-labelledby=""
  className="dialogContentContainer typePopover sizeSmall"
  data-testid="dialog-content-container"
  role="dialog"
/>
`;
